app/uk/gov/hmrc/incometaxsessiondata/controllers/SessionController.scala

```diff
@@ -71,13 +92,66 @@ class SessionController @Inject() (
     }
   }
 
-  private def save(validRequest: SessionData) =
+  private def save(validRequest: Session) =
```

Comment:
Please move all this logic into service layer, not sure why we split 
sessionService.getByMtditid and sessionService.set(validRequest)
It should be much easier to test single service method instead